lib/truffle/truffle/cext.rb

```diff
@@ -70,8 +70,12 @@ def initialize(encoding)
     end
 
     def [](index)
-      raise unless index == NAME_FIELD_INDEX
-      @encoding.name
+      case index
+      when NAME_FIELD_INDEX, 'name'
```

Comment:
Yes - if you access a Ruby object as a C struct, but use a cast to do so and don't assign the casted value to a named local variable or parameter, you get the numerical index rather than the name. This is a Sulong limitation (or bug?).